Add Reve integration notebook

apreshill opened the pull request.

apreshill committed:
add draft reve integration notebook

apreshill committed:
Merge remote-tracking branch 'upstream/main' into reve-nb

apreshill committed:
revisions

apreshill committed:
edits

aaron-siegel committed:
Streamline Makefile (#937)

No one seems to be using or maintaining the Windows hooks in our
Makefile, and they're already broken in several places. This backs them
out so that the Makefile is once again Linux/MacOS only, which
streamlines the Makefile and makes it easier to maintain. If Windows
support becomes important again we'll need to redo it anyway.

(This won't affect any customers using Pixeltable on Windows.)

amithadke committed:
Changes to protocol to handle publishing existing version           (#938)

Adding validation for table.md to have at least one entry

aaron-siegel reviewed (dismissed):
Great overview! Some comments (github won't let me do them inline with the notebook):

- "A Pixeltable runtime (local or remote) with network access to Reve" - it's not clear to me what this means.

- `# Remove the 'reve_demo' directory and its contents, if it exists`
Alternatively just do this in one step:
`pxt.create_dir('reve_demo', if_exists='replace_force')`

- `from pixeltable.functions import reve`
out of place in its own cell; move to the cell where it first gets used

- When you first add the square photo, maybe collect() on all columns to make it clear all the data from both runs is still in the table.

mkornacker committed:
PXT-871: == None filter doesn't work correctly on an array column (#941)

aaron-siegel committed:
More documentation improvements (#936)

- Fix many more broken links
- Faster notebook docs generation
- Add broken-links check to CI

aaron-siegel committed:
Circularity detection in view creation with if_exists='replace' (#942)

Checks for a specific scenario in which:
- A view `my_view` is created
- A subview of `my_view` is created with the identical name `my_view`,
using if_exists='replace'

If this situation not detected, it leads to permanent catalog
corruption.

Xe committed:
Add Tigris integration (#935)

Mostly a copy of the fix in #840 but with Tigris instead of B2. I think
you may need to create a bucket in Tigris with test fixtures. If this is
the case I am more than happy to help!

---------

Signed-off-by: Xe Iaso <xe@tigrisdata.com>

aaron-siegel committed:
Improvements to notebook documentation (#943)

- Fix broken links in notebooks
- Fix markdown errors for Mintlify rendering
- Various Mintlify rendering improvements
- Cleaner build process

sergey-mkhitaryan committed:
Improvements to retriable errors detection in RequestRateScheduler (#922)

With this change, we can detect retry-after inside a TwelveLabs
exception

aaron-siegel committed:
Rename `DataFrame` to `Query` and `DataFrameResultSet` to `ResultSet` (#902)

I tried to be as thorough as I could, renaming local variables and doc
references in addition to classes.

It wasn't entirely trivial, since `df` can also (and often does) mean
"Pandas DataFrame", and also appears as a substring of `udf`, `pdf`, and
`soundfile`. But I did my best to rename carefully.

apreshill committed:
Aaron's fb

apreshill committed:
Merge branch 'main' into reve-nb

apreshill committed:
Merge branch 'main' into reve-nb

apreshill committed:
revised sequencing

apreshill committed:
revised sequencing of notebook

apreshill committed:
Merge branch 'reve-nb' of https://github.com/apreshill/pixeltable into reve-nb

apreshill committed:
update narrative

apreshill committed:
wording

apreshill committed:
Merge remote-tracking branch 'upstream/main' into reve-nb

apreshill committed:
rm update

apreshill committed:
updates with Marcel

apreshill committed:
Squash merge from upstream/main

apreshill committed:
Merge remote-tracking branch 'upstream/main' into reve-nb

apreshill committed:
add doc links

sergey-mkhitaryan commented:
A couple more comments, just my 2c, not necessarily blockers.

> A Reve account with an API key (https://app.reve.com/ → Settings → API Keys)

Looks like they've updated the website, now it's https://app.reve.com/ → API → API Keys

I got confused on step 18 because the `edit` result already has a solarpunk-style background, and I even thought that the images got mixed up, but that's indeed what Reve produces. This makes the further remixing a little bit less interesting. I liked the pipeline you used in the first version I reviewed a little bit more, because you went from the original background to no background (edit) to a specified background (remix).

On step 22 or 23, maybe select and show the original images as well? Otherwise we don't see the original with the tennis player until the recap (step 25).

apreshill committed:
add to integrations nb list in docs.json for mintlify

apreshill committed:
remove bg from edited imgs

apreshill committed:
Merge upstream/main into reve-nb

apreshill committed:
fix parsing errors

apreshill committed:
adding required YAML frontmatter

apreshill committed:
rm description
